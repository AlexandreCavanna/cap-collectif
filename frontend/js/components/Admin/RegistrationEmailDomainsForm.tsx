import React from 'react'
import { connect } from 'react-redux'
import { Row, Col, Button } from 'react-bootstrap'
import { Field, FieldArray, reduxForm, isSubmitting } from 'redux-form'
import renderInput from '../Form/Field'
import { setRegistrationEmailDomains as onSubmit } from '../../redux/modules/user'
import type { State } from '../../types'

const renderDomains = ({
  fields,
}: {
  fields: {
    push: (...args: Array<any>) => any
    map: (...args: Array<any>) => any
    remove: (...args: Array<any>) => any
  }
}): any => (
  <div>
    {fields.map((member, index) => (
      <Row key={index}>
        <Col xs={10}>
          <Field name={`${member}.value`} type="text" component={renderInput} />
        </Col>
        <Col xs={2}>
          <Button onClick={() => fields.remove(index)}>Supprimer</Button>
        </Col>
      </Row>
    ))}
    <Button
      style={{
        marginBottom: 10,
      }}
      onClick={() => fields.push({})}
    >
      Ajouter
    </Button>
  </div>
)

type Props = {
  handleSubmit: (...args: Array<any>) => any
  submitting: boolean
}
export const RegistrationEmailDomainsForm = ({ handleSubmit, submitting }: Props) => (
  <div>
    <p
      style={{
        marginTop: 10,
      }}
    >
      <strong>Nom(s) de domaine</strong>
    </p>
    <form onSubmit={handleSubmit}>
      <FieldArray name="domains" component={renderDomains} />
      <Button type="submit" disabled={submitting}>
        {submitting ? 'Chargement...' : 'Enregistrer'}
      </Button>
    </form>
  </div>
)

const mapStateToProps = (state: State) => ({
  submitting: isSubmitting('registration-email-domains')(state),
  initialValues: {
    domains: state.user.registration_form.domains,
  },
})

const connector = connect(mapStateToProps)
export default connector(
  reduxForm({
    form: 'registration-email-domains',
    onSubmit,
  })(RegistrationEmailDomainsForm),
)
