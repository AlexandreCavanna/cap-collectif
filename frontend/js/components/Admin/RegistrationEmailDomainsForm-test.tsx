/* eslint-env jest */
import * as React from 'react'
import { shallow } from 'enzyme'
import { RegistrationEmailDomainsForm } from './RegistrationEmailDomainsForm'

describe('<RegistrationEmailDomainsForm />', () => {
  const props = {
    submitting: false,
    handleSubmit: jest.fn(),
  }
  it('renders correctly', () => {
    const wrapper = shallow(<RegistrationEmailDomainsForm {...props} />)
    expect(wrapper).toMatchSnapshot()
  })
})
