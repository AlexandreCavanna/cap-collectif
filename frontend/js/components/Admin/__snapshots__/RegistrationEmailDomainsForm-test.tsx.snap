// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RegistrationEmailDomainsForm /> renders correctly 1`] = `
<div>
  <p
    style={
      Object {
        "marginTop": 10,
      }
    }
  >
    <strong>
      Nom(s) de domaine
    </strong>
  </p>
  <form
    onSubmit={[MockFunction]}
  >
    <FieldArray
      component={[Function]}
      name="domains"
    />
    <Button
      active={false}
      block={false}
      bsClass="btn"
      bsStyle="default"
      disabled={false}
      type="submit"
    >
      Enregistrer
    </Button>
  </form>
</div>
`;
